sdk/rust/oak_config_serializer/src/main.rs

```diff
@@ -94,37 +86,55 @@ fn get_sha256(data: &[u8]) -> String {
     hex::encode(hasher.finalize().as_slice().to_vec())
 }
 
-/// Load Wasm module from URL or a file, if URL is not specified.
+/// Download Wasm module from `url` and caches it in `filename`.
```

Comment:
This makes filename both an input and an output, and makes the caching mechanism poorly encapsulated. I would suggest to keep the previous version, and just cache things under a `.oak` did, or directly under `/tmp`, don't know what are the current best practices

Reply:
In fact maybe just under the bin/out dir that we use for final artifacts, add a `oak_module_cache` subfolder or similar

Reply:
Which previous version? From this PR?
Because prior to this PR we didn't cache downloaded files.

Reply:
I meant keep the interface of the previous version, by which either a URL or a filename need to be provided

Reply:
> In fact maybe just under the bin/out dir that we use for final artifacts, add a oak_module_cache subfolder or similar
Currently it saves downloaded files in the `${EXAMPLE}/bin` directory.

Reply:
I thought it downloads them every time though? Caching should be a transparent optimization so that if a file with the correct name and hash is already available, then it won't get downloaded again. I'm not sure that's already the case, if so perhaps there's nothing else to do? :)

Reply:
I mean this PR makes it download a module and save it into `bin`.
And every time it checks whether this file is available in `bin` or not.

Reply:
Ah I see, then perhaps all I'm saying is that it should just hide the filename from the user, and generate one based on just the hash? Think from the point of view of a user, how would they pick a filename? The tool should do as much as possible for them and hide unnecessary details

Reply:
If we want to use `bin` for caching, it should be provided as a parameter to `config_serializer`.
So in order to hide it from the user, I'll just add a `/tmp/oak_config_serializer/${NAME}` directory.

Reply:
We already pass the output file as flag though, right? Maybe we can just create a directory within the target containing directory. Up to you anyways. I am afraid if we just use /tmp it will be hard / impossible to clean up over time

Reply:
Agree, cleaning up `/tmp` is annoying

Reply:
Yet another option: just create a subdirectory under the PWD (which we would just git ignore). This is perhaps the most similar to how git and Bazel and cargo work? Although not exactly

Reply:
If we wanted to make this most similar to cargo, it would have just a flag for the manifest file, and then figure out where to put cached and final artifacts based on that. I guess it would need an extra field in the manifest file for the application name, to be used as file name, if it's not already there. Pretty much the same as rust cargo.toml anyway :)

Reply:
Done
Saving into `.oak/`